### PR TITLE
MWA 2.0 Update

### PR DESCRIPTION
**Pull request by Funkatronics**
Updates Minty Fresh to `clientlib-ktx 2.0` (MWA 2.0 kotlin client lib)

### CODE REVIEW

**Review by ankur2136 — APPROVED** [review]

File: libs/networkInterfaceImpl/build.gradle.kts [hunk 1]
```diff
@@ -16,9 +16,9 @@ android {
         consumerProguardFiles("consumer-rules.pro")
         buildConfigField("String", "SOLANA_RPC_URL", "\"https://api.devnet.solana.com\"")
         buildConfigField(
-            "com.solana.mobilewalletadapter.clientlib.RpcCluster",
-            "RPC_CLUSTER",
-            "com.solana.mobilewalletadapter.clientlib.RpcCluster.Devnet.INSTANCE"
+            "com.solana.mobilewalletadapter.clientlib.Blockchain",
+            "BLOCKCHAIN",
+            "com.solana.mobilewalletadapter.clientlib.Solana.Devnet.INSTANCE"
```
Comment (ankur2136) [inline libs/networkInterfaceImpl/build.gradle.kts:21]: Need to change line 33:35 as well

File: ui/nftMint/src/main/java/com/solanamobile/mintyfresh/nftmint/PerformMintViewModel.kt [hunk 2]
```diff
@@ -69,7 +69,7 @@ class PerformMintViewModel @Inject constructor(
         viewModelScope.launch {
             if (!_viewState.value.isWalletConnected) {
                 val result = mobileWalletAdapter.transact(sender) {
-                    authorize(identityUri, iconUri, identityName, rpcConfig.rpcCluster)
+                    authorize(identityUri, iconUri, identityName, rpcConfig.blockchain.fullName)
```
Comment (ankur2136) [inline ui/nftMint/src/main/java/com/solanamobile/mintyfresh/nftmint/PerformMintViewModel.kt:72]: Is this `authorize` call needed anymore?

File: libs/mintycore/src/main/java/com/solanamobile/mintyfresh/mintycore/usecase/PerformMintUseCase.kt [hunk 3]
```diff
@@ -96,9 +96,10 @@ class PerformMintUseCase @Inject constructor(
 
         val transferResult = walletAdapter.transact(sender) {
             val reauth = reauthorize(identityUri, iconUri, identityName, authToken)
```
Comment (ankur2136) [inline libs/mintycore/src/main/java/com/solanamobile/mintyfresh/mintycore/usecase/PerformMintUseCase.kt:98]: similar comment here. 

`reauthorize` needed? 
transact should handle the auth vs reauth internally right.

File: app/src/main/java/com/solanamobile/mintyfresh/injection/MintyModule.kt [hunk 4]
```diff
@@ -13,7 +18,11 @@ import dagger.hilt.android.components.ViewModelComponent
...
 
     @Provides
-    fun providesMobileWalletAdapter(): MobileWalletAdapter {
-        return MobileWalletAdapter()
+    fun providesMobileWalletAdapter(@ApplicationContext context: Context): MobileWalletAdapter {
+        return MobileWalletAdapter(ConnectionIdentity(
+            identityUri = Uri.parse(context.getString(R.string.id_url)),
+            iconUri = Uri.parse(context.getString(R.string.id_favico)),
+            identityName = context.getString(R.string.app_name)
+        ))
```
[the remark this answers is not in the document]
Comment (Funkatronics) [reply app/src/main/java/com/solanamobile/mintyfresh/injection/MintyModule.kt:26]: no we (Andrew and I) opted to leave the blockchain out as it has a higher chance of changing during usage

File: libs/mintycore/src/main/java/com/solanamobile/mintyfresh/mintycore/usecase/PerformMintUseCase.kt (hunk 3 above)
Comment (Funkatronics) [reply libs/mintycore/src/main/java/com/solanamobile/mintyfresh/mintycore/usecase/PerformMintUseCase.kt:98]: yep ty, cleaned all this up

File: app/src/main/java/com/solanamobile/mintyfresh/injection/MintyModule.kt (hunk 4 above)
[the remark this answers is not in the document]
Comment (Funkatronics) [reply app/src/main/java/com/solanamobile/mintyfresh/injection/MintyModule.kt:26]: my bad, you are totally right here. updated

[the remark this answers is not in the document]
Comment (ankur2136) [reply app/src/main/java/com/solanamobile/mintyfresh/injection/MintyModule.kt]: LGTM

**Pull request merged**